src/queryRewriters/rewriters/patientProxyQueryRewriter.js

```diff
@@ -0,0 +1,166 @@
+                        // replace with patient ids from person
+                        args[`${argName}`] = await async.mapSeries(
+                            argValue,
+                            async a => a.startsWith(patientReferencePlusPersonProxyPrefix) ? await this.getPatientProxyIds(
+                                {
+                                    id: a, databaseQueryManager, includePatientPrefix: true
+                                }) : a
+                        );
+                    }
+                } else if (typeof argValue === 'string' && argValue.startsWith(patientReferencePlusPersonProxyPrefix)) {
+                    args[`${argName}`] = await this.getPatientProxyIds({
+                        id: argValue, databaseQueryManager, includePatientPrefix: true
+                    });
+                }
+            }
+        }
+
+        return args;
+    }
+
+    /**
+     * replaces patient proxy with actual patient ids
+     * @param {string} id
+     * @param {DatabaseQueryManager} databaseQueryManager
+     * @param {boolean} includePatientPrefix
+     * @return {Promise<string>}
+     */
+    async getPatientProxyIds({id, databaseQueryManager, includePatientPrefix}) {
+        // 1. Get person id from id
+        const personId = id.replace(patientReferencePlusPersonProxyPrefix, '').replace(personProxyPrefix, '');
+        // 2. Get that Person resource from the database
+        let patientIds = await this.getPatientIdsFromPersonAsync(
+            {
+                personId,
+                databaseQueryManager
+            }
+        );
+        if (patientIds && patientIds.length > 0) {
+            if (includePatientPrefix) {
+                patientIds = patientIds.map(p => `${patientReferencePrefix}${p}`);
+            }
+            // 4. return a csv of those patient ids (remove duplicates)
+            return Array.from(new Set(patientIds)).join(',');
+        }
+        return id;
+    }
+
+    /**
+     * gets patient ids (recursive) from a person
+     * @param {string} personId
+     * @param {DatabaseQueryManager} databaseQueryManager
+     * @return {Promise<string[]>}
+     */
+    async getPatientIdsFromPersonAsync({personId, databaseQueryManager}) {
+        /**
+         * @type {Person|null}
+         */
+        const person = await databaseQueryManager.findOneAsync(
+            {
+                query: {id: personId}
+            }
+        );
+        /**
+         * @type {string[]}
+         */
+        let patientIds = [];
+        if (person && person.link && person.link.length > 0) {
+            const patientIdsToAdd = person.link
+                .filter(l => l.target.reference.startsWith(patientReferencePrefix))
+                .map(l => l.target.reference.replace(patientReferencePrefix, ''));
+            patientIds = patientIds.concat(patientIdsToAdd);
+            // now find any Person links and call them recursively
+            const personIdsToRecurse = person.link
```

Comment:
I love that we're also dealing with the Person-to-Person links with this. I was worried we'd have to wait until later to add that. This is clean and easy.

Reply:
yep :-)